### ISSUE TITLE
TestIngressGateway503DuringRuleChange flake too frequently

### ISSUE THREAD

**Issue by hzxuzhonghu** [istio/istio#12252]
I have seen this case flake too frequently now in release-1.1 branch.

**Comment by hzxuzhonghu** [comment]
https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12250/istio_auth_sds_e2e/2635

**Comment by esnible** [comment]
I have been looking at this problem a lot.  https://github.com/istio/istio/issues/11700#issuecomment-468709603 and following comments.

The test applies _rule-503test-destinationrule-c-del-subset.yaml_ to eliminate the definitions of v1 and v2.  This *SHOULD NOT* be a problem because the VirtualService is using v3 and v4.

2 seconds later the 503 occurs.

I watched the Envoy logs as I modified a DestinationRule interactively. I saw a 2ms delay between when it logs “gRPC config for type.googleapis.com/envoy.api.v2.Cluster accepted with 82 resources” and when it logs “gRPC config for type.googleapis.com/envoy.api.v2.ClusterLoadAssignment” with the IP and weight of the cluster.
2ms seems very tiny but the test case hits the server every 50ms. If there is a 2ms delay before the subset is configured it would mean a 4% failure rate … which is in the ballpark for what we are all seeing on _TestIngressGateway503DuringRuleChange_.

Suppose when Envoy receives the 82 new resources it throws away or somehow forgets the IP addresses or weights of existing clusters?

I would hope that the IP addresses and weights of the cluster is in the same packet as the list of resources but perhaps it is in the next packet, or perhaps the code that processes the packet is blocked before assigning weights to the revised clusters.

Perhaps Envoy can’t route to a cluster until it has received a ClusterLoadAssignment?
